apps/desktop-wallet/src/api/queries/tokenQueries.ts

```diff
@@ -16,48 +16,45 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { client, ONE_DAY_MS, TOKENS_QUERY_LIMIT } from '@alephium/shared'
+import { batchers, ONE_DAY_MS, throttledClient } from '@alephium/shared'
 import { explorer, NFTMetaData, NFTTokenUriMetaData } from '@alephium/web3'
 import { TokenInfo, TokenStdInterfaceId } from '@alephium/web3/dist/src/api/api-explorer'
 import { queryOptions, skipToken, UseQueryResult } from '@tanstack/react-query'
-import { create, windowedFiniteBatchScheduler } from '@yornaath/batshit'
 import axios from 'axios'
 
+import { SkipProp } from '@/api/apiDataHooks/types'
 import { combineIsLoading } from '@/api/apiDataHooks/utils'
-import { convertDecimalsToNumber, matchesNFTTokenUriMetaDataSchema } from '@/api/utils'
-import { TokenId, UnlistedFT } from '@/types/tokens'
+import { convertTokenDecimalsToNumber, matchesNFTTokenUriMetaDataSchema } from '@/api/utils'
+import { TokenId } from '@/types/tokens'
 
 export type TokenTypesQueryFnData = Record<explorer.TokenStdInterfaceId, TokenId[]>
 
 export const StdInterfaceIds = Object.values(explorer.TokenStdInterfaceId)
 
-interface TokenQueryProps {
+interface TokenQueryProps extends SkipProp {
   id: TokenId
-  skip?: boolean
 }
 
-interface NFTQueryProps {
-  id: TokenId
+interface NFTQueryProps extends TokenQueryProps {
   tokenUri?: NFTMetaData['tokenUri']
-  skip?: boolean
 }
 
 export const tokenTypeQuery = ({ id, skip }: TokenQueryProps) =>
   queryOptions({
     queryKey: ['token', 'type', id],
     queryFn: !skip
       ? async () => {
-          const tokenInfo = await tokenTypeBatchFetcher.fetch(id)
+          const tokenInfo = await batchers.tokenTypeBatcher.fetch(id)
 
           return tokenInfo?.stdInterfaceId
             ? { ...tokenInfo, stdInterfaceId: tokenInfo.stdInterfaceId as TokenStdInterfaceId }
-            : undefined
+            : null
```

Comment:
Tanstack queries should never return `undefined` because that means the query has failed.